### PR TITLE
Az.3.0.0 migration guide

### PR DESCRIPTION
**Pull request by wyunchi-ms**
## Description

Az.3.0.0 migration guide

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,716 @@
...
+  ```powershell
+  $diskconfig = New-AzDiskConfig -Location 'Central US' -UploadSizeInBytes 1023 * 1024 * 1024 * 1024 -SkuName Standard_LRS   -OsType Windows -CreateOption Upload -DiskIOPSReadWrite 500 -DiskMBpsReadWrite 8
+  ```
+	
+### HDInsight
+- `Get-AzHDInsightJobOutput`
+  - Updated the `Get-AzHDInsightJobOutput` cmdlet to support granular role-based access to the storage key.
+  - Users with HDInsight Cluster Operator, Contributor, or Owner roles will not be affected.
+  - Users with only the Reader role will need to specify `DefaultStorageAccountKey` parameter explicitly.
+  #### Users with only Reader role for cmdlet   Get-AzHDInsightJobOutput
```
Comment (dingmeng-xue) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]: Is this format correct?

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,924 @@
...
+```powershell
+Get-AzHDInsightOperationsManagementSuite -Name testcluster
+```
+
+#### After
+```powershell
+Get-AzHDInsightMonitoring -Name testcluster
+```
+
+### `Get-HDInsightProperty`
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
### `Get-AzHDInsightProperty`
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,924 @@
...
+
+#### Before
+Using deprecated alias
+```powershell
+Get-AzHDInsightProperties -Location "East US 2"
+```
+
+#### After
+```powershell
+Get-HDInsightProperty -Location "East US 2"
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
Get-AzHDInsightProperty -Location "East US 2"
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,924 @@
...
+    - [Incompatibility with previous versions of `Az.Resources`](#previous-version-incompatibility-with-az.resources-module)
+  - [Compute](#compute)
+    - [`New-AzDiskConfig`](#new-azdiskconfig)
+  - [HDInsight](#hdinsight)
+    - [`Get-AzHDInsightJobOutput`](#get-azhdinsightjoboutput)
+    - [`Add-AzHDInsightConfigValues`](#add-azhdinsightconfigvalues)
+    - [`Disable-AzHDInsightMonitoring`](#disable-azhdinsightmonitoring)
+    - [`Enable-AzHDInsightMonitoring`](#enable-azhdinsightmonitoring)
+    - [`Get-AzHDInsightMonitoring`](#get-azhdinsightmonitoring)
+    - [`Get-HDInsightProperty`](#get-hdinsightproperty)
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
    - [`Get-AzHDInsightProperty`](#get-azhdinsightproperty)
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,924 @@
...
+#### After
+```powershell
+Remove-AzHDInsightCluster -ClusterName "your-hadoop-001" -PassThru
+True
+```
+
+### `Revoke-AzHDInsightRdpServicesAccess`
+The cmdlet is deprecated. There is no replacement for it.
+
+### `AzHDInsightGatewayCredential`
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
### `Set-AzHDInsightGatewayCredential`
```

Comment (markcowl) [reply documentation/migration-guides/Az.3.0.0-migration-guide.md]: We should add more detail here (can be later this week).  What PowerShell users will be interested in is how the properties have changed,  What properties of the old object are no longer available in the new type, and how can I get equivalents from the new type?

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,924 @@
...
+## RecoveryServices
+
+### `Edit-AzRecoveryServicesAsrRecoveryPlan`
+`ASRRecoveryPlanGroup.ReplicationProtectedItems`, `ASRRecoveryPlanGroup.StartGroupActions` and `ASRRecoveryPlanGroup.EndGroupActions` is removed from output.
+
+### `Get-AzRecoveryServicesAsrRecoveryPlan`
+`ASRRecoveryPlanGroup.ReplicationProtectedItems`, `ASRRecoveryPlanGroup.StartGroupActions` and `ASRRecoveryPlanGroup.EndGroupActions` is removed from output.
+
+### `New-AzRecoveryServicesAsrReplicationProtectedItem`
+Parameter IncludeDiskId is changed in future release to support directly write to managed disk support Azure Site Recovery.
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
Parameter IncludeDiskId is changed to support directly writing to a managed disk in Azure Site Recovery.
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 7]
```diff
@@ -0,0 +1,924 @@
...
+```
+
+
+## Sql
+
+### `Get-AzSqlDatabaseSecureConnectionPolicy`
+Note that secure connection is deprecated and so command is removed. Please use the SQL database blade in the Azure portal to view the connection strings
+
+### `Get-AzSqlDatabaseIndexRecommendations`
+`Get-AzSqlDatabaseIndexRecommendations` alias is removed.
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
`Get-AzSqlDatabaseIndexRecommendations` alias is removed. Use `Get-AzSqlDatabaseIndexRecommendation` instead.
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 8]
```diff
@@ -0,0 +1,924 @@
...
+
+### `Get-AzSqlDatabaseRestorePoints`
+`Get-AzSqlDatabaseRestorePoints` alias is removed.
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
`Get-AzSqlDatabaseRestorePoints` alias is removed. Use `Get-AzSqlDatabaseRestorePoint` instead.
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 9]
```diff
@@ -0,0 +1,924 @@
...
+
+### `Get-AzSqlDatabaseAuditing`
+- The cmdlet `Get-AzSqlDatabaseAudit` is replacing this cmdlet.
+- The output type is changing from the existing type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseBlobAuditingSettingsModel' to the new type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseAuditingSettingsModel'
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
- The output type is changing from the existing type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseBlobAuditingSettingsModel' to the new type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseAuditingSettingsModel', removing properties `AuditState` and `StorageAccountName`. and `StorageAccountSubscriptionId`.  Scripts can retrieve storage account information from the new `StorageAccountResourceId` property. 
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 10]
```diff
@@ -0,0 +1,924 @@
...
+StorageKeyType                      : Primary
+RetentionInDays                     : 0
+EventHubTargetState                 : Enabled
+EventHubName                        : eventHubName
+EventHubAuthorizationRuleResourceId : EventHubAuthorizationRuleResourceId
+LogAnalyticsTargetState             : Enabled
+WorkspaceResourceId                 : "/subscriptions/4b9e8510-67ab-4e9a-95a9-e2f1e570ea9c/resourceGroups/insights-integration/providers/Microsoft.OperationalInsights/workspaces/viruela2"
+```
+
+### `Set-AzSqlDatabaseAudit`
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
### `Set-AzSqlDatabaseAuditing`
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 11]
```diff
@@ -0,0 +1,924 @@
...
+```powershell
+Set-AzSqlDatabaseAuditing -State Enabled -ResourceGroupName "ResourceGroup01" -ServerName "Server01" -StorageAccountName "Storage22" -DatabaseName "Database01"
+```
+
+#### After
+```powershell
+Set-AzSqlDatabaseAudit -ResourceGroupName "ResourceGroup01" -ServerName "Server01" -DatabaseName "Database01" -BlobStorageTargetState Enabled -StorageAccountResourceId "/subscriptions/7fe3301d-31d3-4668-af5e-211a890ba6e3/resourceGroups/resourcegroup01/providers/Microsoft.Storage/storageAccounts/mystorage"
+```
+
+### `Get-AzSqlServerAudit`
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
### `Get-AzSqlServerAuditing`
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 12]
```diff
@@ -0,0 +1,924 @@
...
+- The cmdlet `Get-AzSqlServerAudit` is replacing this cmdlet.
+- The output type is changing from the existing type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseBlobAuditingSettingsModel' to the new type :'Microsoft.Azure.Commands.Sql.Auditing.Model.ServerAuditingSettingsModel'
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
- The output type is changing from the existing type :'Microsoft.Azure.Commands.Sql.Auditing.Model.DatabaseBlobAuditingSettingsModel' to the new type :'Microsoft.Azure.Commands.Sql.Auditing.Model.ServerAuditingSettingsModel'.  Properties `AuditState`, `StorageAccountName`, and `StorageAccountSubscriptionId` are removed.  Scripts that use `StorageAccountName` and `StorageAccountSubscriptionId` proeprties can retrieve this information from the new `StorageAccountResourceId` property.
```

File: documentation/migration-guides/Az.3.0.0-migration-guide.md [hunk 13]
```diff
@@ -0,0 +1,924 @@
...
+StorageKeyType                      : Primary
+RetentionInDays                     : 0
+EventHubTargetState                 : Enabled
+EventHubName                        : eventHubName
+EventHubAuthorizationRuleResourceId : EventHubAuthorizationRuleResourceId
+LogAnalyticsTargetState             : Enabled
+WorkspaceResourceId                 : "/subscriptions/4b9e8510-67ab-4e9a-95a9-e2f1e570ea9c/resourceGroups/insights-integration/providers/Microsoft.OperationalInsights/workspaces/viruela2"
+```
+
+### `Set-AzSqlServerAudit`
```
Comment (markcowl) [inline documentation/migration-guides/Az.3.0.0-migration-guide.md]:
```suggestion
### `Set-AzSqlServerAuditing`
```

**Pull request merged**